driver-core/src/main/java/com/datastax/driver/core/NettyUtil.java

```diff
@@ -0,0 +1,156 @@
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import java.lang.reflect.Constructor;
+import java.util.Locale;
+import java.util.concurrent.ThreadFactory;
+
+import com.google.common.base.Throwables;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A set of utilities related to the underlying Netty layer.
+ */
+@SuppressWarnings("unchecked")
+class NettyUtil {
+
+    private static final boolean FORCE_NIO = SystemProperties.getBoolean("com.datastax.driver.FORCE_NIO", false);
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NettyUtil.class);
+
+    private static final boolean SHADED;
+
+    private static final boolean USE_EPOLL;
+
+    private static final Constructor<? extends EventLoopGroup> EPOLL_EVENT_LOOP_GROUP_CONSTRUCTOR;
+
+    private static final Class<? extends SocketChannel> EPOLL_CHANNEL_CLASS;
+
+    private static final Class[] EVENT_GROUP_ARGUMENTS = { int.class, ThreadFactory.class };
+
+    static {
+        boolean shaded;
+        try {
+            // prevent this string from being shaded
+            Class.forName(String.format("%s.%s.channel.Channel", "io", "netty"));
+            shaded = false;
+        } catch (ClassNotFoundException e) {
+            try {
+                Class.forName("com.datastax.shaded.netty.channel.Channel");
+                shaded = true;
+            } catch (ClassNotFoundException e1) {
+                throw new AssertionError("Cannot locate Netty classes in the classpath:" + e1);
+            }
+        }
+        SHADED = shaded;
+        boolean useEpoll = false;
+        try {
+            Class<?> epool = Class.forName("io.netty.channel.epoll.Epoll");
+            if (FORCE_NIO) {
+                LOGGER.info("Found Netty's native epoll transport in the classpath, "
+                    + "but NIO was forced through the FORCE_NIO system property.");
+            } else if (!System.getProperty("os.name", "").toLowerCase(Locale.US).equals("linux")) {
+                LOGGER.warn("Found Netty's native epoll transport, but not running on linux-based operating " +
+                    "system. Using NIO instead.");
+            } else if (!(Boolean)epool.getMethod("isAvailable").invoke(null)) {
```

Comment:
Oh that's cool!  With this method, we don't need to do the 'os.name' property check anymore, because if Epoll.isAvailable() returns false, we know we're not on a system that supports it.    This is an even better check as it ensures the native libraries are present as well.